src/schemas/index.jsx

```diff
@@ -0,0 +1,27 @@
+import * as Yup from "yup";
+export const signUpSchema = Yup.object({
+  name: Yup.string()
+    .min(3, "Username should be in 3 letter")
+    .max(25)
+    .required("Please Enter Your Username"),
+  password: Yup.string()
+    .min(6, "password must be 6 letter")
+  
+    .required("Please Enter Your Password"),
+  confirm_password: Yup.string()
+    .oneOf([Yup.ref("password"), null], "Passwords don't match")
+    .required("Confirm your password!"),
+});
+export const signInScheema = Yup.object({
+  name: Yup.string()
+    .min(3, "Username should be in 3 letter")
+    .max(25)
+    .required("Please Enter Your Correct Username"),
+  password: Yup.string()
+    .min(6, "password must be 6 letter")
+    .matches(
```

Comment:
and for login you are checking so many things ??
can you explain me this regex??

Reply:
for login you are checking for username and 
password which was generated  by token during the sginup . 
username and password  will be based on the role set during the signup